### PR TITLE
feat(utilities): Add scale_windspeed function from 10m to specified height - Yehui Huang

### LINKED ISSUE #204: Implement equation to scale wind speed
Equation from thermofeel

```
def scale_windspeed(va, h):
    """
    Scaling wind speed from 10 metres to height h
        :param va: (float array) 10m wind speed [m/s]
        :param h: (float array) height at which wind speed needs to be scaled [m]
        returns wind speed at height h
    Reference: Bröde et al. (2012)
    https://doi.org/10.1007/s00484-011-0454-1
    """
    c = 1 / np.log10(10 / 0.01)  #
    c = 0.333333333333
    vh = va * np.log10(h / 0.01) * c

    return vh
```

### PR DESCRIPTION
**Pull request by yehui-h**
## Description
Implemented a new utility function `scale_windspeed` to scale wind speed from 10m reference height to any specified height using the logarithmic wind profile equation. This function is based on the formula from Bröde et al. (2012) and is essential for thermal comfort calculations where wind measurements need to be adjusted to human height levels.

The function accepts both scalar and array inputs for wind speed and height parameters, returning scaled wind speed values as numpy arrays consistent with similar utility functions in the codebase.

No new dependencies required - uses existing NumPy.

Fixes #204

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## How Has This Been Tested?

- [x] Full test suite: Run `tox` to test across all Python versions (3.10-3.13)
- [x] Documentation build: Run `tox -e docs` to verify documentation builds without errors
- [x] Specific Python version: Run `tox -e py312` for targeted testing
- [x] Unit tests: Added 9 test cases in `tests/test_utilities.py` covering scalar inputs, array inputs, edge cases, and physical validation

To reproduce:
```bash
# Clone and checkout branch
git checkout feature/wind-speed-scaling

# Install dependencies
pipenv install --dev

# Run all tests
tox

# Run specific Python version test
tox -e py312

# Build documentation  
tox -e docs
```

**Test Configuration:**

- **Python version:** 3.10, 3.11, 3.12, 3.13
- **Hardware:** N/A (software-only mathematical function)
- **Testing tools:** pytest 8.3.4, tox 4.23.2
- **Dependencies:** NumPy 1.26.4

## Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -1,6 +1,11 @@
 Changelog
 =========
 
+3.4.4 (2025-09-08)
```
Comment (FedericoTartarini) [inline CHANGELOG.rst]: it is great that you included this, very nice that you noticed that. however, I noticed that you did not bump the version of pythermalcomfort

File: docs/documentation/references.rst [hunk 2]
```diff
@@ -18,6 +18,7 @@ References
 .. [Masterson1979] Masterton JM, Richardson FA. Humidex, a method of quantifying human discomfort due to excessive heat and humidity. Downsview, Ontario: CLI 1-79, Environment Canada, Atmospheric Environment Service, 1979
 .. [Havenith2016] Havenith, G., Fiala, D., 2016. Thermal indices and thermophysiological modeling for heat stress. Compr. Physiol. 6, 255–302. DOI: doi.org/10.1002/cphy.c140051
 .. [Blazejczyk2012] Blazejczyk, K., Epstein, Y., Jendritzky, G., Staiger, H., Tinz, B., 2012. Comparison of UTCI to selected thermal indices. Int. J. Biometeorol. 56, 515–535. DOI: doi.org/10.1007/s00484-011-0453-2
+.. [Brode2012] Bröde, P., Fiala, D., Błażejczyk, K., Holmér, I., Jendritzky, G., Kampmann, B., Tinz, B., Havenith, G., 2012. Deriving the operational procedure for the Universal Thermal Climate Index (UTCI). Int. J. Biometeorol. 56, 481–494. DOI: doi.org/10.1007/s00484-011-0454-1
```
Comment (FedericoTartarini) [inline docs/documentation/references.rst]: great that you added the reference here

File: docs/documentation/utilities_functions.rst [hunk 3]
```diff
@@ -38,6 +38,11 @@ Relative air speed
 
 .. autofunction:: pythermalcomfort.utilities.v_relative
 
+Scale wind speed
```
Comment (FedericoTartarini) [inline docs/documentation/utilities_functions.rst:41]: perfect that you added the function to the documentation

File: pythermalcomfort/utilities.py [hunk 4]
```diff
@@ -1245,3 +1245,32 @@ class DefaultSkinTemperature(NamedTuple):
     right_thigh: float = 34.3
     right_leg: float = 32.8
     right_foot: float = 33.3
+
+
+def scale_windspeed(va: float | list[float], h: float | list[float]) -> np.ndarray:
```
Comment (FedericoTartarini) [inline pythermalcomfort/utilities.py]: nice use of tye annotation

File: pythermalcomfort/utilities.py [hunk 5]
```diff
@@ -1245,3 +1245,32 @@ class DefaultSkinTemperature(NamedTuple):
...
+
+    """
+    va = np.array(va)
+    h = np.array(h)
+
+    # Scaling factor: 1/log10(reference_height/roughness_length)
+    c = 1 / np.log10(10 / 0.01)
+    vh = va * np.log10(h / 0.01) * c
+
+    return np.asarray(vh)
```
[the remark this answers is not in the document]
Comment (FedericoTartarini) [reply pythermalcomfort/utilities.py]: I know that z0 was not an input in the example code but I agree with coderabbit and it should have been included as input to the function since the user should be able to change that.

**Review by FedericoTartarini — CHANGES_REQUESTED** [review]
thank you so much for contributing to pythermalcomfort. Very nice work, I left a few comments for you to address

File: CHANGELOG.rst (hunk 1 above)
Comment (yehui-h) [reply CHANGELOG.rst]: Version of pythermalcomfort has been updated in the latest commit

File: pythermalcomfort/utilities.py [hunk 6]
```diff
@@ -1245,3 +1245,84 @@ class DefaultSkinTemperature(NamedTuple):
     right_thigh: float = 34.3
     right_leg: float = 32.8
     right_foot: float = 33.3
+
+
+def scale_windspeed(
+    va: float | list[float], h: float | list[float], z0: float = 0.01
```
Comment (FedericoTartarini) [inline pythermalcomfort/utilities.py]: Thank you for implement this, the value it should not just be a float but the user should also be able to pass an array, as long as they have the same lenght as va and h.

Comment (yehui-h) [reply pythermalcomfort/utilities.py]: Hi @FedericoTartarini , 
support for array z0 inputs in `scale_windspeed` has been implemented.
